File: app/src/main/java/lheureux/votodroid/activities/AddItemActivity.java
package lheureux.votodroid.activities;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

import androidx.appcompat.app.AppCompatActivity;

import lheureux.votodroid.R;
import lheureux.votodroid.databinding.ActivityAddItemBinding;
import lheureux.votodroid.models.Question;
import lheureux.votodroid.services.QuestionService;


public class AddItemActivity extends AppCompatActivity {
    private ActivityAddItemBinding binding;
    private QuestionService questionService;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        questionService = QuestionService.getInstance(this);

        binding = ActivityAddItemBinding.inflate(getLayoutInflater());
        View view = binding.getRoot();
        setContentView(view);

        EditText questionText = findViewById(R.id.editTextTextQuestion);

        binding.submit.setOnClickListener(view1 -> {
            Question question = new Question(questionText.getText().toString());
            try {
                questionService.insert(question);
                Intent intent= new Intent();
                setResult(RESULT_OK, intent);
                finish();
            } catch (Exception e) {
                questionText.setError(e.getMessage());
            }
        });

    }
}

File: app/src/main/java/com/theodorelheureux/votodroid_android/activities/AddItemActivity.kt
package com.theodorelheureux.votodroid_android.activities

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.theodorelheureux.votodroid_android.CreateQuestionMutation
import com.theodorelheureux.votodroid_android.databinding.ActivityAddItemBinding
import com.theodorelheureux.votodroid_android.graphql.GQLClient
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext

class AddItemActivity : AppCompatActivity() {
    private var binding: ActivityAddItemBinding? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityAddItemBinding.inflate(
            layoutInflater
        )
        val view: View = binding!!.root
        setContentView(view)
        val questionText = binding!!.editTextTextQuestion

        binding!!.addItemProgressBar.visibility = View.GONE

        binding!!.submit.setOnClickListener { view1: View? ->

            binding!!.addItemProgressBar.visibility = View.VISIBLE
            binding!!.submit.visibility = View.GONE

            CoroutineScope(Dispatchers.IO).launch {
                var question =
                    GQLClient.instance.mutation(CreateQuestionMutation(questionText.text.toString()))
                        .execute()

                withContext(Dispatchers.Main) {
                    var question = question

                    if (question.data!!.questions.create.errors != null && question.data!!.questions.create.errors!!.isNotEmpty()) {
                        val errors = question.data!!.questions.create.errors!!

                        for (error in errors) {
                            questionText.error = error.message
                        }
                        binding!!.addItemProgressBar.visibility = View.GONE
                        binding!!.submit.visibility = View.VISIBLE
                    } else {
                        val intent = Intent()
                        setResult(Activity.RESULT_OK, intent)
                        finish()
                    }
                }
            }
        }
    }
}